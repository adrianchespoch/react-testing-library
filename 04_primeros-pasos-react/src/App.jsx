import { useState } from 'react';
import PropTypes from 'prop-types';

import './App.css';
import './assets/tnrbjwasofaag8cm7l9v.jpg';
import './assets/react.svg';

function App({ title, initCounter }) {
  const [counter, setCounter] = useState(initCounter);

  // console.log('Render TODO el FC');  // Cuando CAMBIA el STATE se vuelve a Ejecutar el FC

  // // Si la f(x) esta fuera del FC no se reasisgna espacio en memoria con c/rerender
  const handleCounter = () => {
    // setCounter(counter + 1);
    setCounter(c => c + 1);
  };
  const handleSubtract = () => setCounter(counter - 1);
  const handleReset = () => setCounter(initCounter);

  return (
    <div className="App">
      <h1 data-testid="test-title">{title}</h1>
      <h2>{counter}</h2>

      {/* <Counter /> */}
      <div className="card">
        <button className="btn-increase" onClick={handleCounter}>
          +1
        </button>
        <button id="btn-decrease" onClick={handleSubtract}>
          -1
        </button>
        <button aria-label="btn-reset" onClick={handleReset}>
          Reset
        </button>
      </div>

      <p>
        Edit <code>src/App.jsx</code> and save to test HMR
      </p>
      <p className="read-the-docs">
        Click on the Vite and React logos to learn more
      </p>
    </div>
  );
}

export default App;

// // // PropTypes
App.propTypes = {
  title: PropTypes.string.isRequired,
  initCounter: PropTypes.number.isRequired,
};

// // DefaultProps: Entran antes q los PropTypes, evitan error si es  isRequired
App.defaultProps = {
  title: 'No title!',
};
