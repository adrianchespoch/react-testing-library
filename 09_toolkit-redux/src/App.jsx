import { useDispatch, useSelector } from 'react-redux';

import { decrement, increment, incrementByAmount } from './store/slices/counter';
import './App.css';

function App() {
  // Extraemos lo q definimos en el initState del slice
  const { counter } = useSelector(state => state.counter);

  // ya sabe a q reducer va a llegar
  const dispatch = useDispatch();

  return (
    <div className="App">
      <h1>Count is: {counter}</h1>

      <div className="card">
        <button onClick={() => dispatch(increment())}>Increment</button>
        <button onClick={() => dispatch(decrement())}>Decrement</button>
        <button onClick={() => dispatch(incrementByAmount(3))}>Increment by 2</button>
      </div>
    </div>
  );
}

export default App;
