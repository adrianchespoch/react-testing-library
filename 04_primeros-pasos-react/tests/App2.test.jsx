import { render, screen } from '@testing-library/react';
import App from '../src/App';

describe('Pruebas en App', () => {
  const title = 'React Pro';
  const initCounter = 0;

  test('should hacer match con el snapshot', () => {
    const { container } = render(
      <App title={title} initCounter={initCounter} />
    );

    expect(container).toMatchSnapshot();
  });

  test('should mostar el title exacto', () => {
    render(<App title={title} initCounter={initCounter} />);
    // screen.debug();

    expect(screen.getByText(title)).toBeTruthy();
  });

  test('should mostrar el title en 1 h1', () => {
    render(<App title={title} initCounter={initCounter} />);

    expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(
      title
    );
  });

  test('should return value', () => {
    render(<App title={title} initCounter={initCounter} />);

    // expect(screen.getByText(initCounter)).toContain(initCounter);
  });
});
