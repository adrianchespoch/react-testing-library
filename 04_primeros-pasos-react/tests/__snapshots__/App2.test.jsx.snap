// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en App should hacer match con el snapshot 1`] = `
<div>
  <div
    class="App"
  >
    <h1
      data-testid="test-title"
    >
      React Pro
    </h1>
    <p>
      0
    </p>
    <div
      class="card"
    >
      <button>
        +1
      </button>
      <button>
         -1 
      </button>
      <button>
         Reset 
      </button>
    </div>
    <p>
      Edit 
      <code>
        src/App.jsx
      </code>
       and save to test HMR
    </p>
    <p
      class="read-the-docs"
    >
      Click on the Vite and React logos to learn more
    </p>
  </div>
</div>
`;
